src/test/kfs_rest_test.cpp

```diff
@@ -183,7 +183,70 @@ TEST_F(HttpRestApiHandlerTest, dispatchReady) {
     ASSERT_EQ(c, 1);
 }
 
+TEST_F(HttpRestApiHandlerTest, modelMetadataRequest) {
+    std::string request = "/v2/models/dummy/versions/1/";
+    ovms::HttpRequestComponents comp;
+
+    handler->parseRequestComponents(comp, "GET", request);
+    std::string response;
+    handler->dispatchToProcessor(std::string(), &response, comp);
+
+    rapidjson::Document doc;
+    doc.Parse(response.c_str());
+    ASSERT_EQ(std::string(doc["name"].GetString()), "dummy");
+    ASSERT_EQ(std::string(doc["versions"].GetArray()[0].GetString()), "1");
+    ASSERT_EQ(std::string(doc["platform"].GetString()), "OpenVINO");
+
+    ASSERT_EQ(std::string(doc["inputs"].GetArray()[0].GetObject()["name"].GetString()), "b");
+    ASSERT_EQ(std::string(doc["inputs"].GetArray()[0].GetObject()["datatype"].GetString()), "FP32");
+    ASSERT_EQ(doc["inputs"].GetArray()[0].GetObject()["shape"].GetArray()[0].GetInt(), 1);
+    ASSERT_EQ(doc["inputs"].GetArray()[0].GetObject()["shape"].GetArray()[1].GetInt(), 10);
+
+    ASSERT_EQ(std::string(doc["outputs"].GetArray()[0].GetObject()["name"].GetString()), "a");
+    ASSERT_EQ(std::string(doc["outputs"].GetArray()[0].GetObject()["datatype"].GetString()), "FP32");
+    ASSERT_EQ(doc["outputs"].GetArray()[0].GetObject()["shape"].GetArray()[0].GetInt(), 1);
+    ASSERT_EQ(doc["outputs"].GetArray()[0].GetObject()["shape"].GetArray()[1].GetInt(), 10);
+}
+
+TEST_F(HttpRestApiHandlerTest, inferRequestWithMultidimensionalMatrix) {
```

Comment:
Either I am not seeing something here or this should be negative predict request?

Model input shape is [1,10] while we send [5,2]?

Reply:
O I see - this is not going through to gRPC handling & validation. Can we fix shape here in request to match data?